homeassistant/components/fitbit/sensor.py

```diff
@@ -432,48 +366,55 @@ def __init__(
                     measurement_system = FITBIT_MEASUREMENTS["en_US"]
             unit_type = measurement_system[split_resource[-1]]
         self._unit_of_measurement = unit_type
-        self._state = 0
+        self._state: str | None = None
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return self._name
 
     @property
-    def state(self):
+    def state(self) -> str | None:
         """Return the state of the sensor."""
         return self._state
 
     @property
-    def unit_of_measurement(self):
+    def unit_of_measurement(self) -> str | None:
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
     @property
-    def icon(self):
+    def icon(self) -> str | None:
         """Icon to use in the frontend, if any."""
-        if self.resource_type == "devices/battery" and self.extra:
-            battery_level = BATTERY_LEVELS[self.extra.get("battery")]
-            return icon_for_battery_level(battery_level=battery_level, charging=None)
-        return f"mdi:{FITBIT_RESOURCES_LIST[self.resource_type][2]}"
+        if self.resource_type == "devices/battery" and self.extra is not None:
+            extra_battery = self.extra.get("battery")
+            if extra_battery is not None:
+                battery_level = BATTERY_LEVELS.get(extra_battery)
+                if battery_level is not None:
+                    return icon_for_battery_level(battery_level=battery_level)
+        fitbit_ressource = FITBIT_RESOURCES_LIST.get(self.resource_type)
+        if fitbit_ressource is not None:
+            return f"mdi:{fitbit_ressource[2]}"
+        return None
```

Comment:
We already used this key in constructor, so it should be ok at this point.
```suggestion
        fitbit_ressource = FITBIT_RESOURCES_LIST[self.resource_type]
        return f"mdi:{fitbit_ressource[2]}"
```

Reply:
we still need to check if `fitbit_ressource[2]`, since it could be none:
https://github.com/home-assistant/core/blob/c733c4ee5197b90e5769fef9ad2da8a484b613ad/homeassistant/components/fitbit/const.py#L92

Reply:
mhhh ... ok ... now I sse it is an battery sensor ... will add the battery icon here and those no `None` will be possible 😄